packages/edit-site/src/components/add-new-template/new-template.js

```diff
@@ -125,20 +123,19 @@ export default function NewTemplate( {
 
 			// Set template before navigating away to avoid initial stale value.
 			setTemplate( newTemplate.id, newTemplate.slug );
-			// Switch to edit mode.
-			setCanvasMode( 'edit' );
 
 			// Navigate to the created template editor.
 			history.push( {
 				postId: newTemplate.id,
 				postType: newTemplate.type,
+				canvas: 'edit',
 			} );
 
 			createSuccessNotice(
 				sprintf(
 					// translators: %s: Title of the created template e.g: "Category".
 					__( '"%s" successfully created.' ),
-					title
+					newTemplate.title?.rendered || title
```

Comment:
This is not related to the main bug, but it's a minor change to show the proper new template title and not the `slug`.

Reply:
I removed this from this PR and will just open a small follow up, since this might be included in 6.2 and should have just the fix.